docs/tab-widgets/distributed-trace-receive.asciidoc

```diff
@@ -0,0 +1,162 @@
+// ***************************************************
+
+// tag::net[]
+
+Deserialize the incoming distributed tracing context, and pass it to any of the
+{apm-dotnet-ref}/public-api.html#api-start-transaction[`StartTransaction`] or
+{apm-dotnet-ref}/public-api.html#convenient-capture-transaction[`CaptureTransaction`] APIs --
+all of which have an optional `DistributedTracingData` parameter.
+This will create a new transaction or span as a child of the incoming trace context.
+
+Example starting a new transaction:
+
+[source,csharp]
+----
+var transaction2 = Agent.Tracer.StartTransaction("Transaction2", "TestTransaction",
+     DistributedTracingData.TryDeserializeFromString(serializedDistributedTracingData));
+----
+
+// end::net[]
+
+// ***************************************************
+// ***************************************************
+
+// tag::node[]
+
+1. Decode and store the `traceparent` in the receiving service.
+
+2. Pass in the `traceparent` as the `childOf` option to manually start a new transaction
+as a child of the received `traceparent` with
+{apm-node-ref}/agent-api.html#apm-start-transaction[`apm.startTransaction()`].
+
+Example receiving a `traceparent` over raw UDP:
+
+[source,js]
+----
+const traceparent = readTraceparentFromUDPPacket() <1>
+agent.startTransaction('my-service-b-transaction', { childOf: traceparent }) <2>
+----
+<1> Read the `traceparent` from the incoming request.
+<2> Use the `traceparent` to initialize a new transaction that is a child of the original `traceparent`.
+
+// end::node[]
+
+// ***************************************************
+// ***************************************************
+
+// tag::python[]
+
+1. Create a TraceParent object from a string or HTTP header.
+
+2. Start a new transaction as a child of the `TraceParent` by passing in a `TraceParent` object.
+
+Example using HTTP headers:
+
+[source,python]
+----
+parent = elasticapm.trace_parent_from_headers(headers_dict) <1>
+client.begin_transaction('processors', trace_parent=parent) <2>
+----
+<1> Create a TraceParent object from HTTP headers formed as a dictionary
+<2> Begin a new transaction as a child of the received `TraceParent`
+
+TIP: See the {apm-py-ref}/api.html#traceparent-api[`TraceParent` API] for additional examples.
+// end::python[]
+
+// ***************************************************
+// ***************************************************
+
+// tag::ruby[]
+
+Start a new transaction or span as a child of the incoming transaction or span with
+{apm-ruby-ref}/api.html#api-agent-start_transaction[`start_transaction`] or
+{apm-ruby-ref}/api.html#api-agent-start_span[`start_span`].
+
+Example starting a span:
+
+[source,ruby]
+----
+# how to decode `trace_context`?
+
+ElasticAPM.with_transaction 'Do things' do
+  ElasticAPM.start_span 'Do one of the things' # How do we add `trace_context` to this?
+   # ... do something
+  ElasticAPM.end_span
+end
+----
+
+// end::ruby[]
```

Comment:
@elastic/apm-agent-ruby -- I'm having trouble figuring things out for the Ruby agent. Can you please take a look at this example and make recommendations?

These docs are for manually propagating the traceparent in a sending service

Reply:
> Instantiate your own trace context from a Rack `env` object, then pass it when creating your transaction:

```ruby
context = ElasticAPM::TraceContext.parse(env: env)
ElasticAPM.with_transaction("Do things", trace_context: context) do
  ElasticAPM.with_span("Do nested thing", trace_context: context) do
    # However, spans will auto-create a child to their parents transaction's
    # TraceContext of none is passed
  end
end
```

Looking on the other languages' docs, I think this part of the docs is for _receiving_ services, though?

Reply:
Oops! You're right. Thanks!